src/BrazeKit-dev.js

```diff
@@ -361,21 +366,84 @@ var constructor = function () {
         }
     }
 
-    function logNonPurchaseCommerceEventWithProducts(event) {
-        var sanitizedProperties = getSanitizedCustomProperties(
-            event.EventAttributes
-        );
-        var productArray = [];
+    function logNonPurchaseCommerceEventWithProducts(mpEvent) {
+        var brazeEvent = {
```

Comment:
I wonder if we can clean this up a bit by just creating a simpler object, `eventAttributes`, and build it within the switch case and then wrap it in the braze Event object in the end.

For example:

```
const eventAttributes = {};

// Switch Case

eventAttributes.promotion = addPromotions(mpEvent)

// At the end

const brazeEvent = {
  EventName: mpEvent.EventName,
  EventAttributes: mergeObjects(
    brazeEvent.EventAttributes,
    sanitizedProperties
  )
}
```

Reply:
I like this suggestion